pkgs/by-name/ks/ksoloti/package.nix

```diff
@@ -0,0 +1,181 @@
+{ ant
+, copyDesktopItems
+, dfu-util
+, fetchpatch
+, fetchurl
+, fetchFromGitHub
+, gcc-arm-embedded-6
+, lib
+, makeDesktopItem
+, makeWrapper
+, stdenv
+, pkg-config
+, systemd
+, unzip
+, zulu
+,
+}:
+let
+  ksolotiVersion = "1.0.12-8";
+  libusbVersion = "1.0.19";
+  chibiOSVersion = "2.6.9";
+
+  src = fetchFromGitHub {
+    owner = "ksoloti";
+    repo = "ksoloti";
+    rev = ksolotiVersion;
+    sha256 = "sha256-UYxCnEZORpfhaHSl6YohuFiQLPoQ1oyJUL9W6Maf8AE=";
+  };
+
+  libusb = stdenv.mkDerivation rec {
+    name = "libusb";
+    version = libusbVersion;
+    src =
+      let
+        minorVersion = builtins.concatStringsSep "." (
+          lib.take 2 (builtins.splitVersion version)
+        );
+      in
+      fetchurl {
+        url = "https://sourceforge.net/projects/libusb/files/libusb-${minorVersion}/libusb-${version}/libusb-${version}.tar.bz2";
+        sha256 = "sha256-bFAsgWAC+Q1PdgUKZCnDp+DYQgQiLL/y3Old13O6aEA=";
+      };
+
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ systemd ];
+
+    patches = [
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/ksoloti/ksoloti/${ksolotiVersion}/platform_linux/src/libusb.stdfu.patch";
+        sha256 = "sha256-bYzcE20fnwBTZUZmYYHgkW7SekO6pgoj6AaTGIw8kqQ=";
+      })
+    ];
+  };
+
+  patched-dfu-util = dfu-util.overrideAttrs (oldAttrs: {
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ libusb ];
+  });
+
+  chibios = stdenv.mkDerivation {
```

Comment:
Maybe these derivations could become their own packages, so they can be used for other things as well?

Reply:
i'm not sure. it's pretty niche and the current chibios version is miles ahead of this one.